File: mxpack-core/src/test/java/org/odenix/mxpack/core/example/SetWriterOptions.java
/*
 * Copyright 2024 the MxPack project authors
 * SPDX-License-Identifier: Apache-2.0
 */
package org.odenix.mxpack.core.example;

import org.odenix.mxpack.core.BufferAllocator;
import org.odenix.mxpack.core.MessageEncoder;
import org.odenix.mxpack.core.MessageWriter;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.channels.WritableByteChannel;

@SuppressWarnings("unused")
class SetWriterOptions extends Example {
  // -8<- [start:snippet]
  void write(WritableByteChannel channel) throws IOException {
    try (var writer = MessageWriter.of(channel, options -> options
        .allocator(BufferAllocator.ofUnpooled()) //(1)
        .writeBufferCapacity(1024 * 8)
        .stringEncoder(MessageEncoder.ofString())
        .identifierEncoder(MessageEncoder.ofString()))
    ) { /* write some values */ }
  }
  // -8<- [end:snippet]

  @Test
  void test() throws IOException {
    write(outChannel);
  }
}

File: mxpack-kotlin/src/test/kotlin/org/odenix/mxpack/kotlin/example/SetWriterOptions.kt
/*
 * Copyright 2024 the MxPack project authors
 * SPDX-License-Identifier: Apache-2.0
 */
package org.odenix.mxpack.kotlin.example

import java.nio.channels.WritableByteChannel
import org.junit.jupiter.api.Test
import org.odenix.mxpack.kotlin.*

class SetWriterOptions : Example() {
  // -8<- [start:snippet]
  fun write(channel: WritableByteChannel) {
    MessageWriter(
      channel,
      allocator = UnpooledBufferAllocator(), //(1)
      writeBufferCapacity = 1024 * 8,
      stringEncoder = StringEncoder(),
      identifierEncoder = StringEncoder()
    ).use { /* write some values */ }
  }
  // -8<- [end:snippet]

  @Test
  fun test() {
    write(outChannel)
  }
}
